### PR TITLE
[DOCS] - Update the topic-level policies

### LINKED ISSUE #8899: retention support per topic
**Is your enhancement request related to a problem? Please describe.**
We have some complex calculations to be performed on specific topics and those calculations are not finishing within the retention period configured. hence we see data loss. since only on few of the topics we need to these calcuations to be performed we dont want to set retention globally for all topics.

**Describe the solution you'd like**
Need the ability to configure message retention to be configurable per topic to override global setting.
**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sijia-w**
Fixes #8899 

Update the description of the newly added topic-level policies in https://github.com/apache/pulsar/issues/2688.

### CODE REVIEW

**Comment by sijia-w** [comment]
@Jennifer88huang Please help review. Thank you.

File: site2/docs/reference-pulsar-admin.md [hunk 1]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
 Usage
 ```bash
 $ pulsar-admin topics remove-message-ttl tenant/namespace/topic
-=======
+
+Options
```
Comment (Jennifer88huang) [inline site2/docs/reference-pulsar-admin.md:2346]: @sijia-w have you built locally to see the changes?

**Review by Jennifer88huang — COMMENTED** [review]
@sijia-w Thank you for your efforts.
Have you checked and added the new subcommands to the `topics` subcommand lists (from From L1769)?

File: site2/docs/reference-pulsar-admin.md (hunk 1 above)
Comment (sijia-w) [reply site2/docs/reference-pulsar-admin.md:2346]: I didn't add 'Options' inside the code block, but trying to fix the issues already existed.

File: site2/docs/reference-pulsar-admin.md [hunk 2]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
 Usage
 ```bash
 $ pulsar-admin topics remove-message-ttl tenant/namespace/topic
-=======
+
+Options  
+```
```
Comment (Jennifer88huang) [inline site2/docs/reference-pulsar-admin.md:2347]:
```suggestion
```
Options 
```

File: site2/docs/reference-pulsar-admin.md (hunk 1 above)
Comment (Jennifer88huang) [reply site2/docs/reference-pulsar-admin.md:2346]: You can refine it as suggested, move "Options" outside the codeblock.

**Comment by Jennifer88huang** [comment]
> @sijia-w Thank you for your efforts.
> Have you checked and added the new subcommands to the `topics` subcommand lists (from From L1769)?

@sijia-w I've added the subcommand list for `topics` in https://github.com/apache/pulsar/pull/9108

File: site2/docs/reference-pulsar-admin.md [hunk 3]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+### `remove-retention`
+Remove the retention policy that is applied to a topic.
+
+Usage
+```bash
+$ pulsar-admin topics remove-retention tenant/namespace/topic
+```
+
+### `get-dispatch-rate`
+Get message-dispatch-rate for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2378]: should remove the `-` here, like "Get message dispatch rate for a topic."

File: site2/docs/reference-pulsar-admin.md [hunk 4]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+Usage
+```bash
+$ pulsar-admin topics set-dispatch-rate tenant/namespace/topic
+```
+
+### `set-dispatch-rate`
+Set message-dispatch-rate for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2386]: should remove the `-` here, like "Set message dispatch rate for a topic."

File: site2/docs/reference-pulsar-admin.md [hunk 5]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+Usage
+```bash
+$ pulsar-admin topics set-dispatch-rate tenant/namespace/topic
+```
+
+### `remove-dispatch-rate`
+Remove message-dispatch-rate for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2394]: should remove the `-` here.

File: site2/docs/reference-pulsar-admin.md [hunk 6]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+### `remove-delayed-delivery`
+Remove the delayed delivery policy for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics remove-delayed-delivery tenant/namespace/topic
+```
+
+### `get-max-producers`
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2425]: can you please check if this is the correct command? according to https://github.com/apache/pulsar/blob/7c09f5ce649edcca0be792198d97573197c5a272/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L162-L164, the command should be `get-maxProducers` / `set-maxProducers` / `remove-maxProducers`

File: site2/docs/reference-pulsar-admin.md [hunk 7]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+Get max-Producers for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2426]: "Get max number of producers for a topic" would be better.

File: site2/docs/reference-pulsar-admin.md [hunk 8]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+```bash
+$ pulsar-admin topics get-max-producers tenant/namespace/topic
+```
+
+### `set-max-producers`
+Set max-Producers for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics set-max-producers tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2438]:
```bash
pulsar-admin topics set-max-producers tenant/namespace/topic --max-producers 3
```
should include parameter `--max-producers`.

File: site2/docs/reference-pulsar-admin.md [hunk 9]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+### `remove-max-producers`
+Remove max-Producers for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics remove-max-producers tenant/namespace/topic
+```
+
+### `get-max-consumers`
+Get max-Consumers for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2450]: Get max number of consumers for a topic

File: site2/docs/reference-pulsar-admin.md [hunk 10]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+```bash
+$ pulsar-admin topics get-max-consumers tenant/namespace/topic
+```
+
+### `set-max-consumers`
+Set max-Consumers for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics set-max-consumers tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2462]: should include parameter `--max-consumers`.

File: site2/docs/reference-pulsar-admin.md [hunk 11]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+### `remove-max-consumers`
+Remove max-Consumers for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics remove-max-consumers tenant/namespace/topic
+```
+
+### `get-max-unacked-messages-per-subscription`
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2473]: I havent found `get-max-unacked-messages-per-subscription`, `set-max-unacked-messages-per-subscription`, `remove-max-unacked-messages-per-subscription`, `get-max-unacked-messages-per-consumer`, `set-max-unacked-messages-per-consumer`, and `remove-max-unacked-messages-per-consumer` in https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java
Please check if these commands are supported already.

File: site2/docs/reference-pulsar-admin.md [hunk 12]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+### `remove-max-unacked-messages-per-consumer`
+Remove maxUnackedMessagesPerConsumer for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics remove-max-unacked-messages-per-consumer tenant/namespace/topic
+```
+
+### `get-compaction-threshold`
+Get compactionThreshold for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2522]: Get compaction threshold for a topic

File: site2/docs/reference-pulsar-admin.md [hunk 13]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+Usage
+```bash
+$ pulsar-admin topics get-compaction-threshold tenant/namespace/topic
+```
+
+### `set-compaction-threshold`
+Set compactionThreshold for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2530]: Set compaction threshold for a topic

File: site2/docs/reference-pulsar-admin.md [hunk 14]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+Usage
+```bash
+$ pulsar-admin topics set-compaction-threshold tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2534]: should include parameter `--threshold` here.

File: site2/docs/reference-pulsar-admin.md [hunk 15]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+### `remove-compaction-threshold`
+Remove compactionThreshold for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics remove-compaction-threshold tenant/namespace/topic
+```
+
+### `get-offload-policies`
+Get the offload policy for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2546]: Get the offload policies for a topic

File: site2/docs/reference-pulsar-admin.md [hunk 16]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+### `set-offload-policies`
+Set the offload policy for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics set-offload-policies tenant/namespace/topic
+```
+
+### `remove-offload-policies`
+Remove the offload policy for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2562]: Remove the offload policies for a topic

File: site2/docs/reference-pulsar-admin.md [hunk 17]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+### `get-offload-policies`
+Get the offload policy for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics get-offload-policies tenant/namespace/topic
+```
+
+### `set-offload-policies`
+Set the offload policy for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2554]: Set the offload policies for a topic

File: site2/docs/reference-pulsar-admin.md [hunk 18]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+Usage
+```bash
+$ pulsar-admin topics set-offload-policies tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2558]: should document these parameters https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L1302-L1346

File: site2/docs/reference-pulsar-admin.md [hunk 19]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+### `remove-offload-policies`
+Remove the offload policy for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics remove-offload-policies tenant/namespace/topic
+```
+
+### `get-inactive-topic-policies`
+Get the inactive policy for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2570]: Get the inactive topic policies on a topic

File: site2/docs/reference-pulsar-admin.md [hunk 20]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+Usage
+```bash
+$ pulsar-admin topics get-inactive-topic-policies tenant/namespace/topic
+```
+
+### `set-inactive-topic-policies`
+Set the inactive policy for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2578]: Set the inactive topic policies on a topic

File: site2/docs/reference-pulsar-admin.md [hunk 21]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+
+Usage
+```bash
+$ pulsar-admin topics set-inactive-topic-policies tenant/namespace/topic
+```
+
+### `remove-inactive-topic-policies`
+Remove the inactive policy for a topic.
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2586]: Remove inactive topic policies from a topic

File: site2/docs/reference-pulsar-admin.md [hunk 22]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+```bash
+$ pulsar-admin topics get-inactive-topic-policies tenant/namespace/topic
+```
+
+### `set-inactive-topic-policies`
+Set the inactive policy for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics set-inactive-topic-policies tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2582]: should document the parameters defined here: https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L1859-L1871

File: site2/docs/reference-pulsar-admin.md [hunk 23]
```diff
@@ -2267,6 +2252,30 @@ Options
...
+```bash
+$ pulsar-admin topics get-backlog-quotas tenant/namespace/topic
+```
+
+### `set-backlog-quota`
+Set a backlog quota policy for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics set-backlog-quota tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2276]: should document the parameters defined here: https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L984-L989

File: site2/docs/reference-pulsar-admin.md [hunk 24]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
-$ pulsar-admin topics remove-deduplication tenant/namespace/topic
+$ pulsar-admin topics get-retention tenant/namespace/topic
+```
+
+### `set-retention`
+Set the retention policy that is applied to a topic.
+
+Usage
+```bash
+$ pulsar-admin topics set-retention tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2366]: should document the parameters defined here: https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L1181-L1188

File: site2/docs/reference-pulsar-admin.md [hunk 25]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+```bash
+$ pulsar-admin topics remove-retention tenant/namespace/topic
+```
+
+### `get-dispatch-rate`
+Get message-dispatch-rate for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics set-dispatch-rate tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2382]: should document the parameters defined here: https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L1431-L1445

File: site2/docs/reference-pulsar-admin.md [hunk 26]
```diff
@@ -2333,17 +2342,252 @@ Remove the message TTL for a topic.
...
+```bash
+$ pulsar-admin topics get-delayed-delivery tenant/namespace/topic
+```
+
+### `set-delayed-delivery`
+Set the delayed delivery policy for a topic.
+
+Usage
+```bash
+$ pulsar-admin topics set-delayed-delivery tenant/namespace/topic
```
Comment (freeznet) [inline site2/docs/reference-pulsar-admin.md:2414]: should document the parameters defined here: https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L1040-L1048

**Review by freeznet — CHANGES_REQUESTED** [review]
Overall the PR is LGTM, but please refer to my reviews to verify 
- if the command supported already, if not, then we should remove the unsupported commands.
- should we document the parameters for `setter` commands?

File: site2/docs/reference-pulsar-admin.md (hunk 6 above)
Comment (sijia-w) [reply site2/docs/reference-pulsar-admin.md:2425]: @freeznet The functions were changed in this file pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java Please take a look. Thank you.

Comment (freeznet) [reply site2/docs/reference-pulsar-admin.md:2425]: @sijia-w yes, but the change has been overwrite by #8003
https://github.com/apache/pulsar/pull/8003/files#diff-98564f4467b3071af3e1b5fc1d5a0703362d8d9487632cecc12c67d90abda4d2

**Comment by sijia-w** [comment]
@freeznet @Jennifer88huang  Updated. Please take a look. Thank you.

File: site2/docs/reference-pulsar-admin.md [hunk 27]
```diff
@@ -2230,20 +2230,28 @@ Options
 |`-l`, `--ledgerId`|The ledger id |0|
 |`-e`, `--entryId`|The entry id |0|
 
-### `get-deduplication`
-Get a deduplication policy for a topic.
+### `enable-deduplication`
+Enable a deduplication policy for a topic.
```
Comment (Jennifer88huang) [inline site2/docs/reference-pulsar-admin.md:2234]: @what's the differences between `enable-deduplication` and `get-deduplication-enabled`?

File: site2/docs/reference-pulsar-admin.md [hunk 28]
```diff
@@ -2423,167 +2423,167 @@ $ pulsar-admin topics remove-delayed-delivery tenant/namespace/topic
...
+Get the maximum number of unacked messages per subscription for a topic.
 
 Usage
 ```bash
 $ pulsar-admin topics get-max-unacked-messages-per-subscription tenant/namespace/topic
 ```
 
-### `set-max-unacked-messages-per-subscription`
-Set maxUnackedMessagesPerSubscription for a topic.
+### `set-max-unacked-messages-on-subscription`
```
Comment (Jennifer88huang) [inline site2/docs/reference-pulsar-admin.md:2481]: `set-max-unacked-messages-per-subscription`

File: site2/docs/reference-pulsar-admin.md [hunk 29]
```diff
@@ -2422,28 +2422,28 @@ Usage
 $ pulsar-admin topics remove-delayed-delivery tenant/namespace/topic
 ```
 
-### `get-max-producers`
+### `get-maxProducers`
```
Comment (Jennifer88huang) [inline site2/docs/reference-pulsar-admin.md:2425]: The correct form is :`get-max-producers`

File: site2/docs/reference-pulsar-admin.md (hunk 28 above)
Comment (sijia-w) [reply site2/docs/reference-pulsar-admin.md:2481]: @freeznet Can you double confirm the changes?

File: site2/docs/reference-pulsar-admin.md (hunk 29 above)
Comment (sijia-w) [reply site2/docs/reference-pulsar-admin.md:2425]: @freeznet Can you double confirm the changes?

File: site2/docs/reference-pulsar-admin.md (hunk 27 above)
Comment (sijia-w) [reply site2/docs/reference-pulsar-admin.md:2234]: @freeznet I am not sure. Can you take a look at this question. Thank you.

File: site2/docs/reference-pulsar-admin.md (hunk 29 above)
Comment (freeznet) [reply site2/docs/reference-pulsar-admin.md:2425]: @sijia-w just as jennifer mentioned, there are some code errors should be fixed first. such as `get-max-producers` should be added to topic commands.

**Comment by Jennifer88huang** [comment]
@sijia-w It's time to update the docs now since https://github.com/apache/pulsar/pull/9215 is merged. Thank you.

**Pull request closed without merging**